### ISSUE TITLE
Level up system 

### ISSUE THREAD

**Issue by DrkSephy** [DrkSephy/Tales-of-Kratos#4]
In order for the player to grow stronger, an experience system will need to be implemented which allows the character to level up, gain higher stats and progress further in the game.

**Comment by DrkSephy** [comment]
To do this, we'd go about it as shown below:
1. Add an `expGiven` property to `goomba.js`. This will represent the value of experience awarded to the player for killing a goomba.
2. Within `goomba.js`, we create a method called `kill( )`, which will be called when this entity is killed (`health` <= 0). 
3. The `kill( )` method will have code similar to this:
   
   ```
   var player = this.getEntitiesByType( EntityPlayer )[0];
   
   // Award player the experience gained by killing the Goomba
   player.totalExp += this.expGained;
   
   // Decrease experience to next level
   player.nextExp -= this.expGained;
   
   // Call a level up function from main.js
   ig.game.levelUp(player); 
   ```
4. Within `main.js`, we need to write a function which will take the player as a parameter, and check if `nextExp <=0`, which signifies a level up. If this condition is true, we then want to add some values to the player's stats `(def, str, mag), etc`.
5. On future iterations, we'd like to add stats with some random modifiers to make things more interesting. We'd also like to implement some sort of experience gained function which will take in parameters such as the `level` of the enemy, `type` of enemy, etc, instead of having the player gain 50 experience at a time.

**Comment by DrkSephy** [comment]
Implemented by #60.
